lib/archethic/p2p/message/validate_smart_contract_call.ex

```diff
@@ -187,4 +185,18 @@ defmodule Archethic.P2P.Message.ValidateSmartContractCall do
       valid?: valid?
     }
   end
+
+  defp maybe_execute_trigger(contract = %Contract{triggers: triggers}, transaction, opts) do
+    if Map.has_key?(triggers, :transaction) do
+      case Contracts.execute_trigger(:transaction, contract, transaction, [transaction], opts) do
```

Comment:
As discussed I might refactor `execute_trigger` in an other PR to avoid this check every time.